File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxLen = 0;
        unordered_set<char> uniqueCharacters;
        int left = 0;
        int right = 0;
        int n = s.size();
        while(right < n){
            if(uniqueCharacters.find(s[right]) == uniqueCharacters.end()){
                uniqueCharacters.insert(s[right]);
                maxLen = max(maxLen, right-left+1);
                right++;
            }
            else{
                uniqueCharacters.erase(s[left]);
                left++;
            }
        }
        return maxLen;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int left = 0;
        int right = 0;
        int n = s.length();
        int maxLen = 0;
        Set<Character> uniqueCharacters = new HashSet<>();
        while(right < n){
            if(!uniqueCharacters.contains(s.charAt(right))){
                uniqueCharacters.add(s.charAt(right));
                maxLen = Math.max(maxLen, right-left+1);
                right++;
            }
            else{
                uniqueCharacters.remove(s.charAt(left));
                left++;
            }
        }
        return maxLen;
    }
}